### DISCUSSION TITLE
KaTeX extensions

### DISCUSSION THREAD

**Discussion by vangberg** [marimo-team/marimo#2561] [Ideas]
I would like to use [KaTeX extensions](https://katex.org/docs/libs), specifically [mhchem](https://github.com/KaTeX/KaTeX/blob/main/contrib/mhchem/README.md). This requires adding a `<script>` tag to `<head>`.

1. Is there a way to do that?
2. Maybe a general mechanism for specifying KaTeX extensions to load could be a good idea.

**Comment by mscolnick** [comment]
1. Not currently. You could make achieve this with an [anywidget](https://github.com/manzt/anywidget) that writes custom javascript to inject the script. We have had requests to add custom content in the `<head/>` we could do something similar to what we do for css
2. Im unsure how common this would be, but we could include them (and lazy-load them) in the core code, with something like `mo.katex.enabled_ext(["one", "two"])`
